plugins/org.eclipse.n4js.transpiler.es/src/org/eclipse/n4js/transpiler/es/transform/ModuleWrappingTransformation.xtend

```diff
@@ -330,7 +331,11 @@ class ModuleWrappingTransformation extends Transformation {
 			return moduleSpecifierAdjustment.prefix + '/' + module.moduleSpecifier
 
 		var specifier = completeModuleSpecifier
-		val depProject = n4jsCore.findProject(module.eResource.URI).orNull
+		var depProject = n4jsCore.findProject(module.eResource.URI).orNull
+		if (depProject !== null && depProject.projectType === ProjectType.DEFINITION && depProject.definesPackage !== null) {
+			depProject = n4jsCore.findAllProjectMappings.get(depProject.definesPackage);
```

Comment:
The results of `n4jsCore.findAllProjectMappings` could also be cached as part of the `TranspilerState`. This would assure that we only compute the map once per transpiled resource.

Reply:
Thanks, I also think we should do this in a follow-up sub-task or task!